fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -0,0 +1,249 @@
+                .map(s -> (SlotReference) s).collect(Collectors.toSet());
+        for (SlotReference slotReference : slotSet) {
+            String colName = slotReference.getName();
+            if (colName == null) {
+                // TODO: should we throw an exception here?
+                continue;
+            }
+            ColumnStats columnStats = tableStats.getColumnStats(colName);
+            slotToColumnStats.put(slotReference, columnStats);
+        }
+        long rowCount = tableStats.getRowCount();
+        StatsDeriveResult stats = new StatsDeriveResult(rowCount,
+                new HashMap<>(), new HashMap<>());
+        stats.setSlotRefToColumnStatsMap(slotToColumnStats);
+        return stats;
+    }
+
+    private StatsDeriveResult computeAggregate(Aggregate aggregate) {
+        List<Expression> groupByExprList = aggregate.getGroupByExpressionList();
+        StatsDeriveResult childStats = groupExpression.getCopyOfChildStats(0);
+        Map<Slot, ColumnStats> childSlotColumnStatsMap = childStats.getSlotRefToColumnStatsMap();
+        long resultSetCount = -1;
+        for (Expression expression : groupByExprList) {
+            List<SlotReference> slotRefList = expression.collect(SlotReference.class::isInstance);
+            // TODO: Support more complex group expr.
+            //       For example:
+            //              select max(col1+col3) from t1 group by col1+col3;
+            if (slotRefList.size() != 1) {
+                continue;
+            }
+            SlotReference slotRef = slotRefList.get(0);
+            ColumnStats columnStats = childSlotColumnStatsMap.get(slotRef);
+            if (resultSetCount == -1) {
+                resultSetCount = 1;
+                resultSetCount *= columnStats.getNdv();
+            } else if (resultSetCount == 0) {
+                break;
+            }
+        }
+        Map<Slot, ColumnStats> slotColumnStatsMap = new HashMap<>();
+        List<NamedExpression> namedExpressionList = aggregate.getOutputExpressionList();
+        // TODO: 1. Estimate the output unit size by the type of corresponding AggregateFunction
+        //       2. Handle alias, literal in the output expression list
+        for (NamedExpression namedExpression : namedExpressionList) {
+            if (namedExpression instanceof SlotReference) {
+                slotColumnStatsMap.put((SlotReference) namedExpression, new ColumnStats());
+            }
+        }
+        StatsDeriveResult statsDeriveResult = new StatsDeriveResult();
+        statsDeriveResult.setRowCount(resultSetCount);
+        statsDeriveResult.setSlotRefToColumnStatsMap(slotColumnStatsMap);
+        // TODO: Update ColumnStats properly, add new mapping from output slot to ColumnStats
+        return statsDeriveResult;
+    }
+
+    // TODO: Update data size
+    private StatsDeriveResult computeProject(Project project) {
```

Comment:
Stats should be updated according to the project expressions? e.g., ` select a + 1 as b`,  min/max value should be updated accordingly?

Reply:
Will do this later.